### PR TITLE
test separate MailSender 3rd party

### PR DESCRIPTION
**Pull request by julschong**

### CODE REVIEW

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test:acceptance

**Comment by julschong** [comment]
ci:stop

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test:acceptance

**Comment by julschong** [comment]
rebased and resolve conflict in [LPS-152081 Auto, semantic versioning](https://github.com/julschong/liferay-portal/pull/418/commits/86aa7360b8162d3adeb2c157f1c0259d5ece54ae)

**Comment by catarineaguiar** [comment]
some failures were found:

`WikiSubscriptionLocalizedContentTest`, `BlogsEntryLocalServiceTest`, `UserServiceWhenPortalSendsPasswordEmailTest`, `SharingUserNotificationTest`, `MBSubscriptionRootContainerModelTest`. `JournalSubscriptionRootContainerModelTest`
```
[exec] com.liferay.wiki.subscription.test.WikiSubscriptionLocalizedContentTest > testSubscriptionLocalizedContentWhenAddingBaseModel FAILED
     [exec]     java.lang.AssertionError: [] expected:<1> but was:<0>
     [exec]     	at org.junit.Assert.fail(Assert.java:89)
     [exec]     	at org.junit.Assert.failNotEquals(Assert.java:835)
     [exec]     	at org.junit.Assert.assertEquals(Assert.java:647)
     [exec]     	at com.liferay.subscription.test.util.BaseSubscriptionLocalizedContentTestCase.testSubscriptionLocalizedContentWhenAddingBaseModel(BaseSubscriptionLocalizedContentTestCase.java:83)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```
full log [here](https://testray.liferay.com/reports/production/logs/2022-06/test-1-30/test-portal-acceptance-pullrequest(master)/3642/modules-integration-mysql57-jdk8/3/24/jenkins-console.txt.gz?authuser=0)

`TaskNodeExecutorTest`
```
   [exec] com.liferay.portal.workflow.kaleo.internal.runtime.node.test.TaskNodeExecutorTest > testExecuteTimerNotifications FAILED
     [exec]     java.lang.IndexOutOfBoundsException: There are no messages in the inbox
     [exec]     	at com.liferay.portal.test.mail.MailServiceTestUtil.getLastMailMessage(MailServiceTestUtil.java:64)
     [exec]     	at com.liferay.portal.workflow.kaleo.internal.runtime.node.test.TaskNodeExecutorTest.testExecuteTimerNotifications(TaskNodeExecutorTest.java:199)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```
full log [here](https://testray.liferay.com/reports/production/logs/2022-06/test-1-30/test-portal-acceptance-pullrequest(master)/3642/modules-integration-mysql57-jdk8/3/1/jenkins-console.txt.gz?authuser=0)

File: modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/MailSenderImpl.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+	}
+
+	@Override
+	public String getOSGiServiceIdentifier() {
+		return MailSender.class.getName();
+	}
+
+	@Override
+	public Session getSession() {
+		long companyId = CompanyThreadLocal.getCompanyId();
```
Comment (julschong) [inline modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/MailSenderImpl.java:82]: maybe pass in companyId as param

**Comment by julschong** [comment]
need revise

**Pull request closed without merging**